### ISSUE TITLE
Doesn't work with django templates

### ISSUE THREAD

**Issue by ajurna** [cdgriffith/Box#13]
would be nice to use this in django templates. it doesnt resolve correctly when iterating over items.

**Comment by cdgriffith** [comment]
Issue confirmed, testable with:

```python
import django
from django.core.management import settings
from django.conf import global_settings
from django.template import Context, Template

from addict import Dict
from box import Box

b = {
    'BACKEND': 'django.template.backends.django.DjangoTemplates',
    'DIRS': [],
}
global_settings.TEMPLATES.append(b)
settings.configure()
django.setup()

t = Template("Testing {{ item.test_value }}.")

sample_dict = {'item': {'test_value': 3}}
dict_result = t.render(Context(sample_dict))
assert dict_result == "Testing 3.", "Dict failed, was {}".format(dict_result)

adict = Dict(sample_dict)
adict_result = t.render(Context(adict))
assert adict_result == "Testing 3.", "addict failed, was {}".format(
    adict_result)

abox = Box(sample_dict)
abox_result = t.render(Context(abox))
assert abox_result == "Testing 3.", "Box failed, was {}".format(abox_result)
# AssertionError: Box failed, was Testing .
```

**Comment by cdgriffith** [comment]
The django documentation does not match up with what is happening in practice. https://docs.djangoproject.com/en/1.11/ref/templates/api/#variables-and-lookups

Instead of trying a lookup first, it tires if it is callable. Because Box 2.x is callable, it will silently fail. The 3.x branch will remove `__call__` from the box object so that this is not an issue.

**Comment by cdgriffith** [comment]
Fixed with 3.0 https://github.com/cdgriffith/Box/commit/6d90e4213365b66585de60676d317a80fe2607bc
